test/u_FileWatcher.cpp

```diff
@@ -0,0 +1,136 @@
+#define CATCH_CONFIG_RUNNER
+#include "catch.hpp"
+
+#include <algorithm>
+#include <functional>
+#include <limits.h>
+#include <string>
+
+static std::string programmName;
+int main(int argc, char *argv[]) {
+  programmName = argv[0];
+  int result = Catch::Session().run(argc, argv);
+
+  return (result < 0xff ? result : 0xff);
+}
+
+#include "../includes/FileSystemWatcher.h"
+#include "../includes/transforms/AbstractTransform.h"
+
+#include "test_helper.h"
+
+std::string getDirectoryFromFile(const std::string &path) {
```

Comment:
should be `static` (i.e. not part of the public interface of this compilation unit)